app/services/foreman/access_control.rb

```diff
@@ -29,6 +29,21 @@ def permissions
         @permissions
       end
 
+      def permissions_for_controller_action(controller_action)
+        controller_action = path_hash_to_string(controller_action) if controller_action.is_a?(Hash)
```

Comment:
`controller_action` is always a hash isn't it? In any case, this method is introduced in this PR and it's always with the same argument, `path_to_authenticate`. `path_to_authenticate` is always a hash.

How about removing the `controller_action` argument altogether and just calling `path_to_authenticate` here?